numpy/_core/src/multiarray/mapping.c

```diff
@@ -1242,17 +1242,16 @@ array_assign_boolean_subscript(PyArrayObject *self,
 
         NPY_cast_info_xfree(&cast_info);
         if (!NpyIter_Deallocate(iter)) {
-            res = -1;
-        }
-        if (res == 0 && !(flags & NPY_METH_NO_FLOATINGPOINT_ERRORS)) {
-            int fpes = npy_get_floatstatus_barrier((char *)self);
-            if (fpes && PyUFunc_GiveFloatingpointErrors("cast", fpes) < 0) {
-                return -1;
+            if (res ==0) {
+                res = -101;
```

Comment:
Looks mangled up, probably just a typo.  I suppose `-1` as error already set makes this OK, although I wonder if we shouldn't just change the order or skip the check if the deallocate fails (even if I guess that might give the less relevant error).

Reply:
I guess the best thing to do if the deallocate fails is to return -1 immediately, overriding any potential existing PyErr. But what does that mean in terms of calling into python with potentially a PyErr set?

Reply:
You can call `PyErr_Clear` to clear the error indicator if you're sure that's correct and it's set. Might need to wrap in an `if (PyErr_Occurred())` block before calling it.